src/modules/job-info/job_state.c

```diff
@@ -1454,6 +1411,28 @@ static int job_events_alloc (struct job_state_ctx *jsctx,
                                       timestamp);
 }
 
+static int finish_context_parse (flux_t *h,
+                                 struct job *job,
+                                 json_t *context)
+{
+    int status;
+
+    if (!context
+        || json_unpack (context, "{ s:i }", "status", &status) < 0) {
+        flux_log (h, LOG_ERR, "%s: finish context invalid: %ju",
+                  __FUNCTION__, (uintmax_t)job->id);
+        return -1;
```

Comment:
ensure errno gets set since `eventlog_restart_parse()` expects it.  Audit all the other context_parse functions.